File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> mp;
        int left = 0;
        int maxLen = 0;
        int maxFreq = 0;

        for(int right = 0;right<s.length();right++){
            mp[s[right]]++;

            maxFreq = max(maxFreq, mp[s[right]]);

            while((right-left+1) - maxFreq>k){
                mp[s[left]]--;
                left ++;
            }
            maxLen = max(maxLen,right-left+1);
        }
        return maxLen;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] count = new int[26];
        int l = 0;
        int maxFreq = 0;
        int maxLen = 0;
        for (int r = 0; r < s.length(); r++) {
            count[s.charAt(r) - 'A']++;

            maxFreq = Math.max(maxFreq, count[s.charAt(r) - 'A']);

            while (((r - l + 1) - maxFreq) > k) {
                count[s.charAt(l) - 'A']--;
                l++;
            }
            maxLen = Math.max(maxLen, r - l + 1);
        }
        return maxLen;
    }
}
